### PR TITLE
Features/set home locations

### PR DESCRIPTION
**Pull request by DavidBetteridge**
In order to write the tests,  I had to amend the Ghosts to expose their GoHomeStrategies and along with their home locations.

An alternative would be to add HomeX and HomeY properties directly to the Ghost.

For the real game, I haven't checked which Ghost heads to which corner.

### CODE REVIEW

File: src/NPacMan.Game/Game.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public Game(IGameClock gameClock, IGameSettings settings)
         public static Game Create()
         {
             var board = @" XXXXXXXXXXXXXXXXXXXXXXXXXXXX
- X............XX............X
+ Xb...........XX...........iX
```
Comment (DavidBetteridge) [inline src/NPacMan.Game/Game.cs:40]: I haven't checked which Ghost goes where

File: src/NPacMan.Game/Game.cs [hunk 2]
```diff
@@ -65,7 +65,7 @@ X.XX          XX.X
  X......XX....XX....XX......X
  X.XXXXXXXXXX.XX.XXXXXXXXXX.X
  X.XXXXXXXXXX.XX.XXXXXXXXXX.X
- X..........................X
+ Xp........................cX
```
Comment (pgillett) [inline src/NPacMan.Game/Game.cs:68]: The ghost home position is removing a pill/coin

File: src/NPacMan.Game/GameSettingsLoader.cs [hunk 3]
```diff
@@ -49,13 +55,13 @@ public static IGameSettings Load(string board)
                             pacMan = new PacMan(columnNumber - 1, rowNumber, Direction.Left, PacManStatus.Alive, 3);
                             break;
                         case 'X':
-                            walls.Add((columnNumber-1, rowNumber));
+                            walls.Add((columnNumber - 1, rowNumber));
                             break;
                         case '.':
```
Comment (pgillett) [inline src/NPacMan.Game/GameSettingsLoader.cs:60]: We can either include cases for ghost home positions, or simply add the home positions to the coin list

Comment (DavidBetteridge) [reply src/NPacMan.Game/GameSettingsLoader.cs]: Good point - should be sorted now.

File: src/NPacMan.Game/Game.cs (hunk 1 above)
Comment (DavidBetteridge) [reply src/NPacMan.Game/Game.cs]: Fixed

File: src/NPacMan.Game/Ghost.cs [hunk 4]
```diff
@@ -2,30 +2,35 @@
...
-        private readonly IGhostStrategy _homeStrategy;
         public string Name { get; }
         public int X { get; }
         public int Y { get; }
         public IGhostStrategy Strategy { get; }
+        public IGhostStrategy CurrentStrategy { get; }
+        private IGhostStrategy _homeStrategy { get; }
 
-        public Ghost(string name, int x, int y, IGhostStrategy strategy, IGhostStrategy homeStrategy)
+        public int HomeLocationX => ((MoveHomeGhostStrategy)_homeStrategy).X;
```
Comment (DavidBetteridge) [inline src/NPacMan.Game/Ghost.cs:12]: This code only exists as an inspection point for tests

Comment (kevbite) [reply src/NPacMan.Game/Ghost.cs:12]: Maybe the Home Location should just be set on the Ghost? My thoughts are is that we pass in ghosts to the `IGhostStrategy` on the `Move(Ghost, Game)`

Comment (DavidBetteridge) [reply src/NPacMan.Game/Ghost.cs:12]: Yeah,  I was thinking something similar.  Our current design might be over complicating things as well :-)

Comment (kevbite) [reply src/NPacMan.Game/Ghost.cs:12]: Yeah, maybe a little over-engineered... I think it didn't help that we didn't fully understand the requirements for the ghosts originally.

**Pull request merged**